packages/gatsby/src/query/query-runner.ts

```diff
@@ -157,10 +156,18 @@ export const queryRunner = async (
     resultHashes.set(queryJob.id, resultHash)
 
     if (queryJob.isPage) {
-      const publicDir = path.join(program.directory, `public`)
-      const { pages } = store.getState()
-      const page = pages.get(queryJob.id)
-      await pageDataUtil.write({ publicDir }, page, result)
+      // We need to save this temporarily in cache because
+      // this might be incomplete at the moment
+      // since it will not contain included static queries yet
+      // Those will be added later and that is when this file will
+      // be written to `public`
+      const resultPath = path.join(
+        program.directory,
+        `.cache`,
+        `json`,
+        `${queryJob.id.replace(/\//g, `_`)}.json`
```

Comment:
Potentially use a clean identifier here?

Reply:
I think this is fine and implementation detail that shouldn't leak because it will stay in `.cache`.

Reply:
Yeah, that's fair